src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -128,6 +131,7 @@ public class ScriptHost : JobHost
             EventManager = eventManager;
 
             _settingsManager = settingsManager ?? ScriptSettingsManager.Instance;
+            _proxyClient = proxyClient;
```

Comment:
How are you handling changes to the proxy.json file? See ScriptHost.OnChanged - we have logic in place to restart the host when certain files change.